src/test/java/jenkins/plugins/slack/EmailSlackUserIdResolverTest.java

```diff
@@ -100,6 +108,22 @@ public void testResolveUserIdForUser() throws Exception {
         assertEquals(EXPECTED_USER_ID, userId);
     }
 
+    @Test
+    public void testResolveUserIdForUserWithoutResolver() throws Exception {
+
+        PowerMockito.when(MailAddressResolver.resolve(any(User.class))).thenReturn(EMAIL_ADDRESS);
```

Comment:
better to use dependency injection to control this than powermock

Reply:
What would be the type to inject ? `MailAddressResolver` is an abstract class and cannot be instantiated. 

It could be a `java.util.function.Function<User, string>`. What do you think ?

Reply:
The function should work fine